qdrant_client/qdrant_fastembed.py

```diff
@@ -0,0 +1,287 @@
+            batch_size (int, optional):
+                How many documents to encode and upload in single request. Defaults to 32.
+            embedding_model (str, optional):
+                Which embedding model to use. Defaults to "fast-all-MiniLM-L6-v2".
+
+        Raises:
+            ImportError: If fastembed is not installed.
+
+        Returns:
+            List[str]: List of UUIDs of added documents. UUIDs are randomly generated on client side.
+
+        """
+
+        # check if we have fastembed installed
+        embeddings = self._embed_documents(
+            documents=documents,
+            embedding_model_name=embedding_model,
+            batch_size=batch_size,
+        )
+
+        if metadata is None:
+            metadata = [{} for _ in range(len(documents))]
+        else:
+            assert len(metadata) == len(
+                documents
+            ), f"metadata length mismatch: {len(metadata)} != {len(documents)}"
+
+        payloads = ({"document": doc, **metadata} for doc, metadata in zip(documents, metadata))
+
+        if ids is None:
+            ids = [uuid.uuid4().hex for _ in range(len(documents))]
+
+        embeddings_size, distance = self._get_model_params(model_name=embedding_model)
+
+        try:
+            collection_info = self.get_collection(collection_name=collection_name)
+        except Exception:
+            self.create_collection(
+                collection_name=collection_name,
+                vectors_config=models.VectorParams(size=embeddings_size, distance=distance),
+            )
+            collection_info = self.get_collection(collection_name=collection_name)
+
+        assert not isinstance(
+            collection_info.config.params.vectors, dict
+        ), f"Collection have incompatible vector params: {collection_info.config.params.vectors}"
+
+        assert (
+            embeddings_size == collection_info.config.params.vectors.size
+        ), f"Embedding size mismatch: {embeddings_size} != {collection_info.config.params.vectors.size}"
+
+        assert (
+            distance == collection_info.config.params.vectors.distance
+        ), f"Distance mismatch: {distance} != {collection_info.config.params.vectors.distance}"
+
+        self.upload_collection(
+            collection_name=collection_name,
+            vectors=embeddings,
+            payload=payloads,
+            ids=ids,
+            wait=True,
+            **kwargs,
+        )
+
+        return ids
+
+    def query(
+        self,
+        collection_name: str,
+        query_text: str,
+        query_filter: Optional[models.Filter] = None,
+        search_params: Optional[models.SearchParams] = None,
+        limit: int = 10,
+        offset: int = 0,
+        with_payload: Union[bool, Sequence[str], models.PayloadSelector] = True,
+        with_vectors: Union[bool, Sequence[str]] = True,
+        score_threshold: Optional[float] = None,
+        embedding_model: str = DEFAULT_EMBEDDING_MODEL,
```

Comment:
The `query` endpoint should NOT accept `embedding_model`!

This can lead to oversight e.g. folks trying to use different models for `add` and `query`. This means we've to persist the state across multiple client sessions. One way to do so would be to have that as part of the payload itself perhaps?